### ISSUE TITLE
Memory leak, when counter LOG is not displayed

### ISSUE THREAD

**Issue by Simonidzze** [javamelody/javamelody#744]
Hello. I have a problem with memory leak, when I set parametr displayed-counter = ""
```
<context-param>
    <param-name>javamelody.displayed-counters</param-name>
    <param-value></param-value>
</context-param>
```
My application have a many differents Warning and Errors logs, and about 2 weeks it used all memory ~ 10Gb.

I find your class Collector, which must clear the counter, but it's do it, if counter.isDisplayed()
```
Class Collector { // NOPMD

for (final Counter counter : counters) {
	// counter.isDisplayed() peut changer pour spring, ejb, guice ou services selon l'utilisation
	dayCountersByCounter.get(counter).setDisplayed(counter.isDisplayed());
	// collecte pour chaque compteur (hits par minute, temps moyen, % d'erreurs système)
	// Rq : il serait possible d'ajouter le débit total en Ko / minute (pour http)
	// mais autant monitorer les vrais débits réseaux au niveau de l'OS
	if (counter.isDisplayed()) {
		// si le compteur n'est pas affiché (par ex ejb), pas de collecte
		// et pas de persistance de fichiers jrobin ou du compteur
		memorySize += collectCounterData(counter);
	}
}
```

But method initLogs(); calls unconditional in class FilterContext
```
class FilterContext {

FilterContext(final String applicationType) {
		super();
		assert applicationType != null;
		this.applicationType = applicationType;

		boolean initOk = false;
		this.timer = new Timer("javamelody"
				+ Parameters.getContextPath(Parameters.getServletContext()).replace('/', ' '),
				true);
		try {
			logSystemInformationsAndParameters();

			initLogs();
```

I applied temporary solution and set log-transform-pattern= .*
After that memory leak is stopped. But it's a bad solution
I think need clear LOG-counter in any case, or calls initLogs() if counter.isDisplayed()

**Comment by Simonidzze** [comment]
@evernat Hello, Why you closed this issue?

**Comment by evernat** [comment]
I don't know why you would set display-counters to "".
Anyway it's fixed by 1f620f9 and ready for the next release (1.73).
A snapshot build is available at:
https://javamelody.ci.cloudbees.com/job/javamelody/net.bull.javamelody$javamelody-core/

**Comment by Simonidzze** [comment]
Thanks for the quick response
Our monitoring system use only <javaInformation> tag in xml response "http://host:port/monitroing?format=xml&period=jour"
And other information is not required.
When Log_Countr is Displayed, response of "http://host:port/monitroing?format=xml&period=jour" is sometimes equal to 15mb.
